vsphere/internal/virtualdevice/virtual_machine_cdrom_subresource.go

```diff
@@ -438,9 +473,10 @@ func (r *CdromSubresource) Read(l object.VirtualDeviceList) error {
 	if !ok {
 		return fmt.Errorf("device at %q is not a virtual CDROM device", l.Name(d))
 	}
-	if backing, ok := device.Backing.(*types.VirtualCdromIsoBackingInfo); ok {
-		// Only read backing info if it's available. If not, this is a host
-		// device/client mapping that is not managed by TF and needs to be deleted.
+	// Only read backing info if it's available.
+	if _, ok := device.Backing.(*types.VirtualCdromRemoteAtapiBackingInfo); ok {
+		r.Set("client_device", true)
+	} else if backing, ok := device.Backing.(*types.VirtualCdromIsoBackingInfo); ok {
```

Comment:
As above, let's try to avoid `else if` as it's not necessarily good Go.

Instead, write this as a type `switch`:

```go
switch backing := device.Backing.(type) {
case *types.VirtualCdromRemoteAtapiBackingInfo:
  r.Set("client_device", true)
case *types.VirtualCdromIsoBackingInfo:
  ...
default:
  return nil, fmt.Errorf("device at %q is not a virtual CDROM device", l.Name(d))
}
```